runtime/src/main/java/io/micronaut/jackson/ObjectMapperFactory.java

```diff
@@ -119,6 +118,20 @@ public ObjectMapper objectMapper(@Nullable JacksonConfiguration jacksonConfigura
                 targetType.ifPresent(aClass -> module.addDeserializer(aClass, deserializer));
             }
         }
+
+        for (KeyDeserializer keyDeserializer : keyDeserializers) {
+            Class<? extends KeyDeserializer> type = keyDeserializer.getClass();
+            Type annotation = type.getAnnotation(Type.class);
+            if (annotation != null) {
+                Class[] value = annotation.value();
+                for (Class clazz : value) {
+                    module.addKeyDeserializer(clazz, keyDeserializer);
+                }
+            } else {
+                Optional<Class> targetType = GenericTypeUtils.resolveSuperGenericTypeArgument(type);
```

Comment:
This logic doesn't make sense for this case because `KeyDeserializer` doesn't support a generic type argument

Reply:
got it. Removed